web/src/app/services/user.service.ts

```diff
@@ -39,20 +39,27 @@ export class UserService {
     );
   }
 
-  getUserSocialDetails(provider: string, uid: string): Observable<UserSocial> {
-    return from(this.userSocial.ref.where(`${provider}.userId`, '==', uid).get())
+  getUserSocialDetails(uid: string): Observable<DocumentData> {
+    return from(this.userSocial.ref.where('uid', '==', uid).get())
       .pipe(
         map(response => this.getSocialDataFromPayload(response)),
         catchError(error => this.errorService.logError(error))
       );
   }
 
-  getSocialDataFromPayload(response): UserSocial | null {
+  getSocialDataFromPayload(response): DocumentData | null {
     let social;
     if (!response) {
       return null;
     }
-    social = response.docs.pop().data();
+    if (response.empty) {
+      return response;
+    }
+    social = {
+      social: response.docs.pop().data(),
```

Comment:
Does the data not contain the ID already?

Reply:
no the data doesn't contain the document Id. Will create a new model for socialDoc and will use that.

Reply:
Ok, do we not have an existing model we can use?

Reply:
We should have a domain model (core models), we always use

Reply:
As of now I haven't yet created a domain model that handles the document metadata and the data simultaneously. We have only userSocial model which is the document data model. I will modify/create a domain model that handles the above scenario.